internal/app/go-filecoin/paymentchannel/manager.go

```diff
@@ -236,6 +218,105 @@ func PaychActorCtorExecParamsFor(client, miner address.Address) (initActor.ExecP
 	return p, nil
 }
 
+// GetMinerWorkerAddress mocks getting a miner worker address from the miner address
+func (pm *Manager) GetMinerWorkerAddress(ctx context.Context, miner address.Address, tok shared.TipSetToken) (address.Address, error) {
+	view, err := pm.stateViewer.GetStateView(ctx, tok)
+	if err != nil {
+		return address.Undef, err
+	}
+	_, fcworker, err := view.MinerControlAddresses(ctx, miner)
+	return fcworker, err
+}
+
+func (pm *Manager) WaitForCreatePaychMessage(ctx context.Context, mcid cid.Cid) (address.Address, error) {
+	var newPaychAddr address.Address
+
+	handleResult := func(b *block.Block, sm *types.SignedMessage, mr *vm.MessageReceipt) error {
+		var res initActor.ExecReturn
+		if err := encoding.Decode(mr.ReturnValue, &res); err != nil {
+			return err
+		}
+
+		var msgParams initActor.ExecParams
```

Comment:
Why are we decoding this and ctorParams? It doesn't appear to be used?

Reply:
good catch; this was moved from the old code. I might have been using it to get the From and To addresses before.